metricbeat/module/system/process_summary/process_summary.go

```diff
@@ -0,0 +1,97 @@
+// +build darwin freebsd linux windows
+
+package process_summary
+
+import (
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/metricbeat/mb/parse"
+	"github.com/elastic/beats/metricbeat/module/system/process"
+	"github.com/pkg/errors"
+
+	sigar "github.com/elastic/gosigar"
+)
+
+// init registers the MetricSet with the central registry.
+// The New method will be called after the setup of the module and before starting to fetch data
+func init() {
+	if err := mb.Registry.AddMetricSet("system", "process_summary", New, parse.EmptyHostParser); err != nil {
```

Comment:
So the panic is caused by a naming collision the happens when registering the metrics key for `metricbeat.system.process.summary` with libbeat monitoring. Since the `process` metricset is already registered we end up with a collision. We could work around the issue with a change to this [line](https://github.com/andrewkroh/beats/blob/d3d14072be53041835210ebba4d65bfc54bed204/metricbeat/mb/module/wrapper.go#L362-L362) to remove dots from the key name. This would "flatten" the metrics reported about the metricsets and simplify consuming those metrics in any sort of UI.

    key := fmt.Sprintf("metricbeat.%s.%s", module, strings.Replace(name, ".", "_", -1))

But @tsg's concern that the usage of dots is overloaded was spot on.